packages/kit/src/vite/index.js

```diff
@@ -341,22 +361,47 @@ function kit() {
 
 			if (svelte_config.kit.prerender.enabled) {
 				// this is necessary to close any open db connections, etc.
-				// TODO: prerender in a subprocess so we can exit in isolation
+				// TODO: prerender in a subprocess so we can exit in isolation and then remove this
 				// https://github.com/sveltejs/kit/issues/5306
 				process.exit(0);
 			}
 		},
 
+		/**
+		 * Adds the SvelteKit middleware to do SSR in dev mode.
+		 * @see https://vitejs.dev/guide/api-plugin.html#configureserver
+		 */
 		async configureServer(vite) {
 			return await dev(vite, vite_config, svelte_config);
 		},
 
+		/**
+		 * Adds the SvelteKit middleware to do SSR in preview mode.
+		 * @see https://vitejs.dev/guide/api-plugin.html#configurepreviewserver
+		 */
 		configurePreviewServer(vite) {
 			return preview(vite, svelte_config, vite_config.preview.https ? 'https' : 'http');
 		}
 	};
 }
 
+/** @param {import('rollup').OutputBundle} bundle */
+function collect_output(bundle) {
+	/** @type {import('rollup').OutputChunk[]} */
+	const chunks = [];
+	/** @type {import('rollup').OutputAsset[]} */
+	const assets = [];
+	for (const key of Object.keys(bundle)) {
+		// collect asset and output chunks
+		if (bundle[key].type === 'asset') {
+			assets.push(/** @type {import('rollup').OutputAsset} */ (bundle[key]));
+		} else {
+			chunks.push(/** @type {import('rollup').OutputChunk} */ (bundle[key]));
+		}
+	}
```

Comment:
The key isn't used directly, might as well refactor it to Object.values while we're here.

Reply:
Yep, we can also leverage TS control flow for discriminated unions and remove the type assertion.